src/web-ui/src/public/components/AmazonPayButton.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+  <div id="AmazonPayButton"></div>
+</template>
+
+<script>
+import {RepositoryFactory} from "@/repositories/RepositoryFactory";
+
+const CartsRepository = RepositoryFactory.get('carts')
+
+export default {
+    name: "AmazonPayButton",
+    data () {
+      return {
+        payload: {
+          "webCheckoutDetails": {
+            // This is a truncated demo: instead of continuing the session, we route to the Amazon Pay instructions.
+            // To enable the full process we need to handle the order review and downstream processing and
+            // a full HTTPS service on the Retail Demo Store.
+            "checkoutReviewReturnUrl": 'https://developer.amazon.com/docs/amazon-pay-checkout/introduction.html'
+          },
+          "storeId": process.env.VUE_APP_AMAZON_PAY_STORE_ID,
+          "chargePermissionType": "OneTime"
+        },
+        payloadSignature: null
+      }
+    },
+    mounted () {
+      this.loadAmazonPayButton()
+    },
+    methods: {
+        async loadAmazonPayButton() {
+            let recaptchaScript = document.createElement('script')
+            recaptchaScript.setAttribute('src', 'https://www.google.com/recaptcha/api.js')
+            document.head.appendChild(recaptchaScript)
```

Comment:
What is the purpose of this?

Reply:
This has been fixed now. New code:

            if (!document.getElementById("amazon-pay-checkout-javascript")) {
              let payscript = document.createElement('script')
              payscript.setAttribute('src', 'https://static-na.payments-amazon.com/checkout.js')
              payscript.setAttribute('id', 'amazon-pay-checkout-javascript')
              document.head.appendChild(payscript)
            }